### PR TITLE
Remove rules about CN-ID

### PR DESCRIPTION
**Pull request by richsalz**
Don't use the CN-ID as an identifier for server identity.

### CODE REVIEW

File: draft-ietf-uta-rfc6125bis.md [hunk 1]
```diff
@@ -593,10 +577,10 @@ TLS server:
...
 Most security-related terms in this document are to be understood in
 the sense defined in {{SECTERMS}}; such terms include, but are not
-limited to, "attack", "authentication", "authorization",
-"certification authority", "certification path", "certificate",
-"credential", "identity", "self-signed certificate", "trust", "trust
-anchor", "trust chain", "validate", and "verify".
+limited to, "attack," "authentication," "authorization,"
+"certification authority," "certification path," "certificate,"
+"credential," "identity," "self-signed certificate," "trust," "trust
+anchor," "trust chain," "validate," and "verify".
```
Comment (briansmith) [inline draft-ietf-uta-rfc6125bis.md:583]: Is this intended to be in this PR? It looks unrelated. I see there are a lot of changes to quotes in this PR. Perhaps we should do those changes in a separate PR that focuses on such reformatting, and then rebase this on top of that or vice versa.

File: draft-ietf-uta-rfc6125bis.md [hunk 2]
```diff
@@ -921,19 +823,13 @@ document.
...
-  the server's fully qualified DNS domain name in a CN-ID.
-  Therefore, the certificate SHOULD NOT include a CN-ID unless the
-  certification authority issues the certificate in accordance with a
-  specification that reuses this one and that explicitly encourages
-  continued support for the CN-ID identifier type in the context of a
-  given application technology.
+5. Certification authorities
+  SHOULD NOT issue certificates that represent
+  the server's fully qualified DNS domain name in a Common Name
+  RDN of the Subject Name.
```
Comment (briansmith) [inline draft-ietf-uta-rfc6125bis.md:829]: As was discussed previously on the mailing list, recommending or disallowing CAs to include CN-IDs in the subject CN is a separate issue from requiring relying parties to only use subjectAltName when validating a certificate for a DNS name.

At our strictest, I think we should continue to allow CAs to include any DNS name in subject CN that also appears (verbatim) in the subjectAltName as a dNSName or iPAddress.

Or, perhaps we just don't make any recommendations at all regarding the contents of the subject CN. The main purpose of revising this RFC is to make the subject CN irrelevant as far as certificate verification is concerned.

File: draft-ietf-uta-rfc6125bis.md [hunk 3]
```diff
@@ -1138,49 +1026,35 @@ in accordance with the following rules:
...
+types); it only needs to construct the functional equivalent of such identifiers
+for matching purposes.
 
 ### Examples {#verify-reference-examples}
 
 A web browser that is connecting via HTTPS to the website at `www.example.com`
-might have two reference identifiers: a DNS-ID of `www.example.com` and,
-as a fallback, a CN-ID of `www.example.com`.
+might have two reference identifiers: a DNS-ID of `www.example.com` and
+a DNS-ID of `w3.example.com`.
```
Comment (briansmith) [inline draft-ietf-uta-rfc6125bis.md:1051]: I don't understand how the browser gets "w3.example.com" from "www.example.com" here. Typical operation of a web browser would be to only have "www.example.com," no?

I think in this PR we should stick to removing the CN-ID from this example, changing "two reference identifiers" to "one reference identifier".

If we want to say something about more exotic name resolution stuff for web browsers, we should do that in a separate PR.

File: draft-ietf-uta-rfc6125bis.md [hunk 4]
```diff
@@ -1207,16 +1081,12 @@ its list of reference identifiers without finding a match.  The search succeeds
...
-
-> Security Warning: A client MUST NOT seek a match for a reference identifier
-> of CN-ID if the presented identifiers include a DNS-ID, SRV-ID, URI-ID, or
-> any application-specific identifier types supported by the client.
+Implementation Note: A client might be configured to perform
+multiple searches, i.e., to match more than one reference
+identifier.
+Although such behavior is not forbidden by this specification, rules
+for matching multiple reference identifiers are a matter for
+implementation or future specification.
```
Comment (briansmith) [inline draft-ietf-uta-rfc6125bis.md:1089]: I think this change should be reverted or moved to a formatting-only PR.

**Review by briansmith — COMMENTED** [review]
Some initial comments. I didn't go through the whole updated draft and look for missed changes yet.

File: draft-ietf-uta-rfc6125bis.md (hunk 1 above)
Comment (richsalz) [reply draft-ietf-uta-rfc6125bis.md:583]: Gah, stpuid git mistake.  Will fix in a day or two.

File: draft-ietf-uta-rfc6125bis.md (hunk 2 above)
Comment (richsalz) [reply draft-ietf-uta-rfc6125bis.md:829]: I will remove this item, you're right, focusing on the validation only is simpler.

File: draft-ietf-uta-rfc6125bis.md (hunk 3 above)
Comment (richsalz) [reply draft-ietf-uta-rfc6125bis.md:1051]: Chagned to single reference identifier.

File: draft-ietf-uta-rfc6125bis.md (hunk 4 above)
Comment (richsalz) [reply draft-ietf-uta-rfc6125bis.md]: okay, reverted the formatting change.

File: draft-ietf-uta-rfc6125bis.md (hunk 1 above)
Comment (richsalz) [reply draft-ietf-uta-rfc6125bis.md]: updated commit pushed fixing the errors.

**Comment by richsalz** [comment]
Merged locally and pushed.
